app/src/main/java/com/example/service/data/source/local/SongLocalDataSource.kt

```diff
@@ -0,0 +1,23 @@
+package com.example.service.data.source.local
+
+import android.content.Context
+import android.util.Log
+import com.example.service.data.model.Song
+import com.example.service.data.source.remote.SongDataSource
+
+class SongLocalDataSource( var context: Context) : SongDataSource{
```

Comment:
Thiết kế singleton phải private constructor 
Ưu tiên val hơn var 
Chú ý Ctrl + Alt + L; Ctrl + Alt + O mỗi khi code 
```suggestion
class SongLocalDataSource private constructor(val context : Context) : SongDataSource{
```